server/mongodb/models/BlogPost.js

```diff
@@ -23,7 +23,25 @@ const BlogPostSchema = new Schema({
     type: Date,
     default: Date.now,
   },
+  orderIndex: {
+    type: Number,
+  },
+});
+
+BlogPostSchema.pre("save", function (next) {
```

Comment:
Added pre save hook so that before a blog post is created it populates the `orderIndex` new blog post.
The first index is 0, so we can simply retrieve the current document count for `BlogPost` collection which will be the orderIndex for the new blog.